### PR TITLE
feat: add schema_from_json FFI function

### PR DESCRIPTION
**Pull request by chiinlquah**
## What changes are proposed in this pull request?

Add a new FFI function `schema_from_json` that parses a Delta schema JSON string and returns a `Handle<SharedSchema>`.

```rust
pub unsafe extern "C" fn schema_from_json(
    json: KernelStringSlice,
    allocate_error: AllocateErrorFn,
) -> ExternResult<Handle<SharedSchema>>
```

This allows constructing a schema handle from a JSON representation without needing an existing snapshot. The primary use case is creating new tables where the caller defines the schema in Java/C++ and needs to pass it to `get_create_table_builder`.

### Changes
- Add `schema_from_json` + `schema_from_json_impl` to `ffi/src/lib.rs`
- Add `serde_json` to FFI crate `[dependencies]` (already in `[dev-dependencies]` and transitively available via `delta_kernel`)
- Add 3 tests: valid schema, invalid JSON, empty struct

### CODE REVIEW

File: ffi/src/lib.rs [hunk 1]
```diff
@@ -978,6 +978,31 @@ pub unsafe extern "C" fn free_schema(schema: Handle<SharedSchema>) {
...
+///
+/// This allows constructing a schema handle from a JSON representation (following the
+/// [Delta Protocol schema serialization format](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#schema-serialization-format))
+/// without needing an existing snapshot. Useful for creating new tables with a caller-defined schema.
+///
+/// The returned handle must be freed with [`free_schema`] when no longer needed.
+///
+/// # Safety
+///
+/// Caller is responsible for passing a valid `allocate_error` function.
```
Comment (thinh2) [inline ffi/src/lib.rs:991]: and a valid `json` string.

File: ffi/Cargo.toml
Comment (thinh2) [inline ffi/Cargo.toml:41]: since dep in [dependencies] is used in all binary target,  let's remove the redundant `serve_json` in [dev-dependencies]

**Review by thinh2 — APPROVED** [review]
LGTM with small nits!

**Comment by chiinlquah** [comment]
Discussed offline, we can change the create table builder to consume from EngineSchema instead and use the visitor pattern to convert it into a RustSchema, following the same convention we use for FFI today.
